### PR TITLE
refactor set method

### LINKED ISSUE #241: Refactor set-method
* [x] add a highlevel-functions that allows formatting like the current set-method or better
* [x] add a convenient way to restore a "good" default-value-set (like now by calling `set()`)

### PR DESCRIPTION
**Pull request by patkan**
* Adds a set method that allows to set a particular formatting without affecting the rest.
* Extends the testing.
* Allows for the old behavior with `set_with_default()`: calling without parameters will restore a default value.
* fixes #241
* Also improves the mypy testing.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -76,4 +76,4 @@ all =
 [flake8]
 exclude = .git,.tox,.github,.eggs,__pycache__,doc/conf.py,build,dist,capabilities-data,test,src/escpos/constants.py
 max-line-length = 120
-extend-ignore = E203, W503
+extend-ignore = W503
```
Comment (patkan) [inline setup.cfg:79]:
```suggestion
extend-ignore = E203, W503
```

**Comment by patkan** [comment]
@belono, if you want to, please have a look at these changes. I am changing also that now mypy is called in CI. If you have any objections or issues with that, please let me know.
Thank you!

**Comment by patkan** [comment]
> @belono, if you want to, please have a look at these changes. I am changing also that now mypy is called in CI. If you have any objections or issues with that, please let me know. Thank you!

Hi @belono would it be ok to merge this branch?

**Comment by belono** [comment]
> > @belono, if you want to, please have a look at these changes. I am changing also that now mypy is called in CI. If you have any objections or issues with that, please let me know. Thank you!
> 
> Hi @belono would it be ok to merge this branch?

`mypy` is a great tool that I have never used before and from now on it will be a permanent part of my tooling. It's perfect to have it in CI.

The new implementation of the `set` method is looking really good.
No objections to merge at all. :+1:

**Comment by patkan** [comment]
Great, thanks for having a look!

**Pull request merged**